include/OGL/MatrixWrapper/HostMatrix.H

```diff
@@ -0,0 +1,231 @@
+        const lduInterfaceFieldPtrsList &interfaces_) const;
+
+
+    /** Iterates all interfaces and collect the corresponding cell id (row)
+    ** and a unique counter
+    **
+    ** @return vector of size nnz_non_local, with a running index, row
+    ** index, and a unique compressed column index
+    ** sections for each interface
+    ** ret = [(1,2,1),(2,20, 2), (3, 20, 2) ...]
+    **         i0   i1,   i...
+    */
+    [[nodiscard]] std::vector<std::tuple<label, label, label, label>>
+    collect_cells_on_non_local_interface(
+        const lduInterfaceFieldPtrsList &interfaces) const;
+
+public:
+    // segregated matrix wrapper constructor
+    HostMatrixWrapper(const ExecutorHandler &exec, const objectRegistry &db,
+                      label nrows, label upper_nnz, bool symmetric,
+                      const scalar *diag, const scalar *upper,
+                      const scalar *lower, const lduAddressing &addr,
+                      const FieldField<Field, scalar> &interfaceBouCoeffs,
+                      const FieldField<Field, scalar> &interfaceIntCoeffs,
+                      const lduInterfaceFieldPtrsList &interfaces,
+                      const dictionary &solverControls, const word &fieldName,
+                      label verbose);
+
+    // segregated matrix wrapper constructor
+    HostMatrixWrapper(const ExecutorHandler &exec, const objectRegistry &db,
+                      const lduAddressing &addr, bool symmetric,
+                      const scalar *diag, const scalar *upper,
+                      const scalar *lower,
+                      const FieldField<Field, scalar> &interfaceBouCoeffs,
+                      const FieldField<Field, scalar> &interfaceIntCoeffs,
+                      const lduInterfaceFieldPtrsList &interfaces,
+                      const dictionary &solverControls, const word &fieldName,
+                      label verbose);
+
+    /** Based on OpenFOAMs ldu matrix format this function computes two
+     ** consecutive index arrays (local_sparsisty_.row_idxs and col_idxs) in row
+     ** major ordering and the permutation index (local_sparsity_.ldu_mapping),
+     ** which are required to create to a ginkgo matrix
+     **/
+    [[nodiscard]] std::shared_ptr<SparsityPattern> compute_local_sparsity(
+        std::shared_ptr<const gko::Executor> exec) const;
+
+    /** Copies the LDU matrix coefficients to local_coeffs without changing or
+     ** reinstantiating the sparsity pattern.
+     ** This uses the local_sparsity_.ldu_mapping to permute the data already
+     ** on the host or device to be in row major order.
+     **/
+    void compute_local_coeffs(std::shared_ptr<const SparsityPattern> sparsity,
+                              gko::array<scalar> &target_coeffs) const;
+
+    /** Copies the interface matrix coefficients to non_local_coeffs without
+    ** changing or reinstantiating the sparsity pattern.
+    **/
+    void compute_non_local_coeffs(
```

Comment:
Is `compute_non_local_coeffs` implemented?

Reply:
Same as for the local_coeffs.